app/Actions/DataFixes/ReorderStreamMessages.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Actions\DataFixes;
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Lorisleiva\Actions\Concerns\AsCommand;
+
+class ReorderStreamMessages
+{
+    use AsCommand;
+
+    public string $commandSignature = 'data-fix:reorder-stream-messages';
+
+    public function handle()
+    {
+        $this->backupStreamMessages();
```

Comment:
To be as paranoid as possible, should probably have try/catch around backing up of messages-- if this fails we wouldn't be able to restore, so shouldn't go any further.

Reply:
It occurred to me that if line 17 errors out, the following lines will not run, but in the spirit of your paranoia, I've added a check that the backup table has the same number of records as the original and throws an exception if they're not equal.

I've also wrapped the backup and check in a try/catch block to explicitly handle the case that the backup fails.